src/test_heap.py

```diff
@@ -0,0 +1,296 @@
+# -*- coding: utf-8 -*-
+"""Testing module for the Heap classes."""
+
+from heap import Heap
+import pytest
+import random
+
+
+# ------------------Heap Initialization Tests--------------------
+# [x] Test default initialization of a heap is a heap.
+# [x] Test the inital size of a heap is 0.
+# [x] Test the default initializtion is a max heap.
+# [x] Test the override to a min heap.
+# [x] Test if invalid input into min/max heap results in an error.
+
+def test_heap_initalization():
+    """Test default initialization of a heap is a heap."""
+    heap = Heap()
+    assert isinstance(heap, Heap)
+
+
+def test_heap_size_initally_0():
+    """Test the inital size of a heap is 0."""
+    heap = Heap()
+    assert heap.size == 0
+
+
+def test_heap_init_default_max():
+    """Test the default initializtion is a max heap."""
+    heap = Heap()
+    assert heap._min_max == 'max'
```

Comment:
There's another outcome of this that you could test, give how you've implemented the switch between min and max heaps.  Can you think of how you'd verify that your methods for re-sorting the heap were properly allocated?